### ISSUE TITLE
Ability to run scripts before  running sources

### ISSUE THREAD

**Issue by umasoya** [b4b4r07/afx#6]
# WHAT

I'm using `sindresorhus/pure`. 
I want to load the configuration before running `sources`.

# Workaround

1. Create an empty repository.
2. Run only snippet without loading anything.

```
- name: empty
  description: Empty repo for pure
  owner: umasoya
  repo: empty
  plugin:
    snippet: |
      zstyle :prompt:pure:git:branch color magenta
      zstyle :prompt:pure:git:branch:cached color yellow
      zstyle :prompt:pure:git:dirty color 091
      zstyle :prompt:pure:user color blue
      zstyle :prompt:pure:host color blue

- name: sindresorhus/pure
  description: Pretty, minimal and fast ZSH prompt
  owner: sindresorhus
  repo: pure
  plugin:
    sources:
      - pure.zsh
```

**Comment by babarot** [comment]
Thanks for the suggestion. I added `snippet-prepare` to support this case. This is still on beta so it'll may be changed in near future but basically I want to support this pattern.

**Comment by babarot** [comment]
Merged #10
